packages/api/test/e2e/doughnut-v0.e2e.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019 Centrality Investments Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import testingPairs from '@plugnet/keyring/testingPairs';
+import { Api } from '../../src/Api';
+import { Doughnut } from '@cennznet/types/extrinsic/Doughnut';
+import { hexToU8a, assert } from '@plugnet/util'
+import { cryptoWaitReady } from '@cennznet/util';
+import { EventRecord, Vector } from '@plugnet/types';
+
+describe('Doughnut for CennznetExtrinsic', () => {
+  let api: Api;
+  let keyring;
+
+  beforeEach(async () => {
+    await cryptoWaitReady();
+    if (!api) {
+      // TODO: Change to Rimu when doughnut goes live...
+      // api = await Api.create({provider: 'wss://rimu.unfrastructure.io/public/ws'});
+      api = await Api.create({ provider: 'ws://localhost:9944' });
+      keyring = testingPairs({ type: 'sr25519' });
+    }
+  });
+
+  afterEach(() => {
+    jest.setTimeout(5000);
+  });
+
+  it('It issues a doughnut with GA transfer for CENNZnet domain', async done => {
+    const holder = keyring.bob;
+    const receiver = keyring.charlie;
+    // Doughnut:
+    // issuer: alice, holder bob
+    // module: generic-asset, method: transfer
+    // block cooldown: 100
+    const doughnut = "0x000000d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a481840d0ba63656e6e7a6e657400000000000000009200000000406d6f64756c650000000000000000000000000000000000000000000000000000806d6574686f64000000000000000000000000000000000000000000000000000026000000e401a1cd84f9e4f285ce8071cdb6e1415cfe307616a9257f8cb7c720902c8821bc769bb934cbe3b3f4cf1e5f9c9668b96deaa6379c91f56678015de9dbc8780f";
+    const tx = api.tx.genericAsset.transfer(16001, receiver.address, 10000);
+    tx.addDoughnut(new Doughnut(hexToU8a(doughnut)));
+
+    await tx.signAndSend(holder, ({ _, status }) => {
+      if (status.isFinalized) {
+        done();
```

Comment:
need check for who's paying tx fee, and the Transfer event, who's the from account.

Reply:
I will unit test this rust side and will add some e2e test when we support delegation fee payment as a feature

Reply:
at least we can verify the from Account in Transfer event.

Reply:
reopen here.

Reply:
Do you have an example of checking the transfer event account?

Reply:
When `status.isFinalized` is true, find event.method === "Transfer" in events
```
({ events, status }) => {
  if (status.isFinalized) {
    const transferEvent = events.find(e=>e.method === "Transfer");
...
  }
}
```